### PR TITLE
Add security labels for anonymized resources.

### PR DESCRIPTION
**Pull request by BoyaWu10**

### CODE REVIEW

File: src/Fhir.Anonymizer.Core/AnonymizerEngine.cs [hunk 1]
```diff
@@ -117,29 +126,52 @@ public ElementNode AnonymizeResourceNode(ElementNode root)
...
+            foreach (var resource in resourceNodes)
             {
-                var newResource = AnonymizeResourceNode(GetResourceRoot(resource));
+                var status = new AnonymizationStatus();
+                var anonymizedNode = AnonymizeResourceNode(GetResourceRoot(resource), status);
+                
+                anonymizedNode.RemoveNullChildren();
+                var anonymizedResource = anonymizedNode.ToPoco<Resource>();
+                
+                anonymizedResource.TryAddSecurityLabels(status);
```
Comment (deepak-ms) [inline src/Fhir.Anonymizer.Core/AnonymizerEngine.cs:158]: For a bundle, do we add security label to Bundle resource or each entry resource?

File: src/Fhir.Anonymizer.FunctionalTests/TestResources/patient-basic-target.json [hunk 2]
```diff
@@ -1,6 +1,24 @@
...
+        "code": "REDACTED",
+        "display": "part of the resource is removed"
+      },
+      {
+        "system": "http://terminology.hl7.org/CodeSystem/v3-ObservationValue",
+        "code": "ABSTRED",
+        "display": "exact value is replaced with a range"
+      },
+      {
+        "code": "PERTURBED",
```
Comment (deepak-ms) [inline src/Fhir.Anonymizer.FunctionalTests/TestResources/patient-basic-target.json:17]: Where is "PERTURBED" defined? didn't find it in "https://www.hl7.org/fhir/v3/ObservationValue/cs.html".

File: src/Fhir.Anonymizer.Core/AnonymizerEngine.cs (hunk 1 above)
Comment (BoyaWu10) [reply src/Fhir.Anonymizer.Core/AnonymizerEngine.cs:158]: For a Bundle, if entry resources are anonymized, security labels will both be added to Bundle resource and entry resources. We did not find any specific guidance for this scenario in both the specs of [security labels](https://www.hl7.org/fhir/security-labels.html) and [Bundle](http://hl7.org/fhir/bundle.html). But we find that it is OK to add security labels to Bundle resource (in the description of both spec) and entry resources (in an [example](https://www.hl7.org/fhir/security-labels.html#rsl) of the spec of security labels). Here are some [internal discussion](https://microsofthealth.visualstudio.com/Health/_workitems/edit/72972) around it in the discussion field of this working item.

File: src/Fhir.Anonymizer.FunctionalTests/TestResources/patient-basic-target.json (hunk 2 above)
Comment (BoyaWu10) [reply src/Fhir.Anonymizer.FunctionalTests/TestResources/patient-basic-target.json:17]: Yes, it is a label created by our own, as currently there is no available label to describe the behavior of date shift. Details could be found in the [internal discussion](https://microsofthealth.visualstudio.com/Health/_workitems/edit/72972) as well.

Comment (deepak-ms) [reply src/Fhir.Anonymizer.FunctionalTests/TestResources/patient-basic-target.json:17]: Thanks Boya for the info!

File: src/Fhir.Anonymizer.Core/Models/AnonymizationStatus.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            if (!string.Equals(originalValue, currentValue, StringComparison.InvariantCulture))
+            {
+                IsAbstracted = true;
+            }
+        }
+
+        public void UpdateIsPerturbed(string originalValue, string currentValue)
+        {
+            if (!string.Equals(originalValue, currentValue, StringComparison.InvariantCulture))
```
Comment (moria97) [inline src/Fhir.Anonymizer.Core/Models/AnonymizationStatus.cs:31]: How about dateshift delta equals to zero?

File: src/Fhir.Anonymizer.Core/Processors/KeepProcessor.cs [hunk 4]
```diff
@@ -1,10 +1,11 @@
-﻿using Hl7.Fhir.ElementModel;
+﻿using Fhir.Anonymizer.Core.Models;
+using Hl7.Fhir.ElementModel;
 
 namespace Fhir.Anonymizer.Core.Processors
 {
     public class KeepProcessor: IAnonymizerProcessor
     {
-        public void Process(ElementNode node)
+        public void Process(ElementNode node, AnonymizationStatus status)
```
Comment (skywalker-wu) [inline src/Fhir.Anonymizer.Core/Processors/KeepProcessor.cs:8]: More looks like we are trying to use a input parameter to return some result. Could we change the interface to something like:
public ProcessResult Process(ElementNode node)? And merge result outside of processor?

File: src/Fhir.Anonymizer.Core/Models/AnonymizationStatus.cs (hunk 3 above)
Comment (BoyaWu10) [reply src/Fhir.Anonymizer.Core/Models/AnonymizationStatus.cs:31]: I think we should add security labels in this case as well, as the date is shifted to range [-50, 50] and zero is one of them.

File: src/Fhir.Anonymizer.Core/Processors/KeepProcessor.cs (hunk 4 above)
Comment (BoyaWu10) [reply src/Fhir.Anonymizer.Core/Processors/KeepProcessor.cs]: Yes, it is fixed in the update.

**Pull request merged**